projects/burnr/src/components/NodeSelector.tsx

```diff
@@ -1,34 +1,192 @@
 import React from 'react';
 
-import { Typography } from '@material-ui/core';
```

Comment:
That's a lot of work in here, but I'd actually prefer a styled input `<select>`  in this case, since it's specifically designed for this kind of interaction, it's  accessible by screen readers, it comes with an event handler  and a `selected` prop.

I can try to port over your Styles and the filter function to my branch and see how far I'm getting.

Reply:
Ah, it actually does have an event-trigger and is probably also optimized for screen readers. I didn't look at the MUI Autocomplete component before.

Let's merge it when you're done and then I can try to fiddle my node selection stuff in